File: sakila-payment-service/src/main/java/com/example/sakila/module/rental/event/RentalEventUtils.java
package com.example.sakila.module.rental.event;

import com.example.sakila.module.rental.Rental;
import com.example.sakila.module.rental.event.model.RentalEventDTO;

public class RentalEventUtils {

  public static RentalEventDTO toDTO(Rental rental) {
    RentalEventDTO dto = new RentalEventDTO();

    dto.setId(rental.getId());
    dto.setRentalDate(rental.getRentalDate());
    dto.setFilmId(rental.getFilmId());
    dto.setStoreId(rental.getStoreId());
    dto.setReturnDate(rental.getReturnDate());
    dto.setStaffId(rental.getStaffId());
    dto.setLastUpdate(rental.getLastUpdate());

    if (rental.getCustomer() != null) {
      dto.setCustomerId(rental.getCustomer().getId());
    } else {
      throw new IllegalArgumentException(String.format("Rental with id %d has no customer", rental.getId()));
    }

    return dto;
  }
}

File: sakila-payment-service/src/main/java/com/example/sakila/module/rental/event/RentalEventUtils.kt
package com.example.sakila.module.rental.event

import com.example.sakila.module.rental.Rental
import com.example.sakila.module.rental.event.model.RentalEventDTO

class RentalEventUtils {

  companion object {

    @JvmStatic
    fun toDTO(rental: Rental): RentalEventDTO {
      val dto = RentalEventDTO()

      dto.id = rental.id
      dto.rentalDate = rental.rentalDate
      dto.filmId = rental.filmId
      dto.storeId = rental.storeId
      dto.returnDate = rental.returnDate
      dto.staffId = rental.staffId
      dto.lastUpdate = rental.lastUpdate

      dto.customerId = rental.customer?.id?:
          throw IllegalArgumentException("Rental with ID ${rental.id} has not customer")

      return dto
    }

  }
}
